api/v1alpha2/azuremachine_conversion.go

```diff
@@ -173,6 +173,15 @@ func Convert_v1alpha3_Image_To_v1alpha2_Image(in *infrav1alpha3.Image, out *Imag
 	return nil
 }
 
+// Convert_v1alpha3_IngressRule_To_v1alpha2_IngressRule converts IngressRule from v1alpha3 to v1alpha2
+func Convert_v1alpha3_IngressRule_To_v1alpha2_IngressRule(in *infrav1alpha3.IngressRule, out *IngressRule, s apiconversion.Scope) error { //nolint
```

Comment:
you also need to add conversion for the newly added fields. See https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/729/files#diff-3962b452782abe8db61ec4e1f455e6b0R49 for an example.